### PR TITLE
Proposal for configuring modules through imports

### PR DESCRIPTION
**Pull request by jathak**
This is a very rough draft of the proposal to implicitly create configurations when importing libraries that have already migrated to the module system (see #2772).

Since the semantics are just modifications of the module system spec, I just specified the sections that should be modified and stated their new contents. I'm not sure if it would be better to just directly modify that spec, but this seemed like a significant enough change that it shouldn't go through the fast-track process.

### CODE REVIEW

**Comment by nex3** [comment]
This looks good so far. My main concern with this approach is the potential overhead of creating a new configuration for every `@import`—do you have any thoughts on how we could mitigate that?

**Comment by jathak** [comment]
What if we only created the new configuration if the file being imported contains a `@forward` rule? Would that be sufficient?

**Comment by nex3** [comment]
Yeah, that should work. Good thinking!

That's not something you need to include as spec text, since it's
effectively an implementation detail, but if you want you can add a
non-normative implementation advisory.

**Comment by jathak** [comment]
Okay, I added that note and also clarified that a new config created by a prefixed `@forward` is implicit if the original config was.

I also expanded on the design decisions section. Is there anything else you think I should bring up there?

File: proposal/configuring-modules-through-imports.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+issues if the same library is imported more than once.
+
+While the solution we settled on does not perfectly cover all use cases that
+worked before the library migrated to the module system, we think it strikes a
+good balance of supporting most existing use cases without hurting performance
+or making the language specification and implementation overly complicated.
+
+One potential use case that will still not work is if a downstream user of the
+library attempts to change its configuration between two imports of the same
+library, that change will be ignored. However, this is an edge case that is (a)
```
Comment (nex3) [inline proposal/configuring-modules-through-imports.md]: This first sentence is kind of awkward. Consider breaking it up into two sentences instead.

File: proposal/configuring-modules-through-imports.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+probably not intended by the user, (b) relatively easy to fix by moving all
+declared configuration variables before all library imports, and (c) very
+difficult to support for a library using the module system without compromising
+the module system's [goals][].
```
Comment (nex3) [inline proposal/configuring-modules-through-imports.md]: I suggest being more specific about how it compromises which goals.

**Review by nex3 — COMMENTED** [review]
A couple suggestions, but otherwise looks good!

**Review by nex3 — APPROVED** [review]

**Comment by jathak** [comment]
Can this be merged? (I don't have write access to this repo)

**Pull request merged**